runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java

```diff
@@ -240,6 +264,23 @@ else if (state == GameState.LOGIN_SCREEN)
 		}
 	}
 
+	private void updateAttackStyle(int attackStyleIndex, int castingMode)
+	{
+		AttackStyle[] attackStyles = equippedWeaponType.getAttackStyles();
+		if (attackStyleIndex < attackStyles.length)
```

Comment:
Can this ever be false?

Reply:
Good question. 

I couldn't remember why this was here. But then I realized this function [already exists](https://github.com/runelite/runelite/blob/99a66ccd693d3190c55c14d89b94092de63b01ee/runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java#L257-L272), so that's where it came from. The original was added [seven years ago](https://github.com/runelite/runelite/commit/2a406a017050219cb7375e271928d4103533618f#diff-fe962845d7199590d8b2c8527054d78214cd318b5fdcdaf3a3520836766e8ba8R114-R129) with the first implementation of the Attack Styles plugin (previously called Attack Indicator).

Digging more into the question more... 

- the value of `attackStyles` has to be exactly [4 or 6](https://github.com/runelite/runelite/blob/99a66ccd693d3190c55c14d89b94092de63b01ee/runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/WeaponType.java#L91-L96)

- `attackStyleIndex` is the same as `client.getVarpValue(VarPlayer.ATTACK_STYLE)` (unfortunately I'm not sure what values are possible)